src/Symfony/Component/Security/Http/HttpUtils.php

```diff
@@ -135,6 +135,10 @@ public function generateUri($request, $path)
             return $path;
         }
 
+        if (!$request instanceof Request) {
```

Comment:
This targets 2.7, preventing the fatal error. Making this the first check would break for people which rely on this check. I would suggest to throw an exception here as it is and deprecate passing something else than a Request in 3.3, for adding the typehint in 4.0.